### PR TITLE
`BCE` loss: forward ad support

### PR DESCRIPTION
**Pull request by nikitaved**
As per title + BCE with logits gets a simpler implementation.
Relevant for https://github.com/pytorch/pytorch/issues/71117

### CODE REVIEW

**Comment by soulitzer** [comment]
From the "unexpected success" looks like this also fixed the composite compliance issue?

**Review by soulitzer — APPROVED** [review]
Do you think the formula is simple enough now that we could potentially inline this? See formula for `kl_div` for example.

Otherwise looks good, thanks for the simplification!

**Comment by nikitaved** [comment]
Thank you, @soulitzer . Yes, it could be inlined, just not a fun of super long lines :) Shall I do it?
